istioctl/pkg/util/configdump/cluster.go

```diff
@@ -43,13 +43,10 @@ func (w *Wrapper) GetDynamicClusterDump(stripVersions bool) (*adminapi.ClustersC
 
 // GetClusterConfigDump retrieves the cluster config dump from the ConfigDump
 func (w *Wrapper) GetClusterConfigDump() (*adminapi.ClustersConfigDump, error) {
-	if w.Configs == nil {
-		return nil, fmt.Errorf("config dump has no cluster dump")
-	}
-	clusterDumpAny, ok := w.Configs["clusters"]
-	if !ok {
+	if w.Configs == nil || len(w.Configs) < 3 {
 		return nil, fmt.Errorf("config dump has no cluster dump")
 	}
+	clusterDumpAny := w.Configs[2]
```

Comment:
Clusters are `w.Configs[1]` see the response on your question in the Envoy repo and [here](https://www.envoyproxy.io/docs/envoy/latest/api-v2/admin/v2alpha/config_dump.proto).

Reply:
Maybe also worth linking to those docs in comments at some point in this package to explain the ordering.

Reply:
Ah, right, thanks for catching this, I must confused it in the code.

Another question, I omitted `Bootstrap` in [configdump()](https://github.com/istio/istio/blob/975720d6b932e7d540d585505bb20bd475254793/pilot/pkg/proxy/envoy/v2/ads.go#L243) as it looks doesn't make too much sense to include an empty dump?

Reply:
Oh, ignore my my question, I have to include the bootstrap otherwise it failed the test.